### DISCUSSION TITLE
Request for Integrating Large Language Model File Search API

### DISCUSSION THREAD

**Discussion by xielong** [langgenius/dify#5095] [Suggestion]
I was wondering if Dify supports the File Search feature offered by large language models, like the one OpenAI has.  (see [OpenAI File Search API](https://platform.openai.com/docs/assistants/tools/file-search)).

This isn't about Dify having its own file search, but rather using the file search capabilities of the language model itself. 

It would be super helpful for us to quickly search through multiple documents.

**Comment by patryk20120** [comment]
Well for me is a kind of a RAG, so you can upload the document into knowledge and do the same.

**Reply by qweasdzxcpkh** [reply]
i think it is not same exactly, i see dify's running detail is nesting the knowledge searched context into the prompt.

**Reply by Yash-1511** [reply]
> i think it is not same exactly, i see dify's running detail is nesting the knowledge searched context into the prompt.

When you are using an agent app in dify it will use knowledge as a tool. And openai uses the agent technique which is the same as in the dify agent app.
